aioshelly/coap.py

```diff
@@ -69,6 +86,74 @@ def socket_init():
     return sock
 
 
+def verify_l2socket_creation_permission():
+    """Create a socket using the scapy configured l2socket.
+
+    Try to create the socket
+    to see if we have permissions
+    since sniff will do it another
+    thread so we will not be able to capture
+    any permission or bind errors.
+    """
+    conf.L2socket()
+
+
+class MulticastQuerier:
+    """Multicast querier management."""
+
+    def __init__(self):
+        """Initialize multicast querier thread."""
+        try:
+            verify_l2socket_creation_permission()
+        except (Scapy_Exception, OSError) as ex:
+            if os.geteuid() == 0:
+                _LOGGER.error("Cannot watch for multicast query packets: %s", ex)
+            else:
+                _LOGGER.debug(
+                    "Cannot watch for multicast query packets without root or CAP_NET_RAW: %s",
+                    ex,
+                )
+            return
+        _LOGGER.debug("Multicast querier thread started")
+        self.stop_thread = False
+        self.thread = threading.Thread(target=self.run)
+        self.thread.start()
```

Comment:
We should not start threads in the constructor. Please move this to a start method.
```suggestion

    def start(self):
        """Start the multicast querier."""
        self.thread = threading.Thread(target=self.run)
        self.thread.start()
```

Reply:
Done